tests/lib/nm/wired_test.py

```diff
@@ -113,3 +114,118 @@ def test_get_info_with_invalid_duplex(ethtool_mock, NM_mock):
         'mac-address': dev_mock.get_hw_address.return_value,
         'mtu': dev_mock.get_mtu.return_value
     }
+
+
+class TestWiredSetting(object):
+
+    def test_identity(self):
+        state = {}
+        obj1 = obj2 = nm.wired.WiredSetting(state)
+
+        assert obj1 == obj2
+        assert not (obj1 != obj2)
+
+    def test_empty_state_is_false(self):
+        state = {}
+        obj = nm.wired.WiredSetting(state)
+
+        assert not obj
+
+    def test_no_relevant_keys_is_false(self):
+        state = {'foo': 'boo'}
+        obj = nm.wired.WiredSetting(state)
+
+        assert not obj
+
+    def test_relevant_keys_with_false_values_is_false(self):
+        state = {
+            schema.Interface.MTU: 0,
+            schema.Interface.MAC: '',
+        }
+        obj = nm.wired.WiredSetting(state)
+
+        assert not obj
+
+    def test_partial_relevant_keys_is_true(self):
+        state = {
+            schema.Interface.MTU: 1500,
+            schema.Interface.MAC: 'abc',
+        }
+        obj = nm.wired.WiredSetting(state)
+
+        assert obj
+
+    def test_equality_for_empty_states(self):
+        state = {}
```

Comment:
I wonder if it would be better to use two different dicts here as well...

Reply:
The identity of the dict should not matter.
Different dicts with the same content is checked later on, so I guess it is ok to exercise both.

Reply:
Why should it not matter? You do not seem to trust anything in these tests and the test name says "test equality for *empty* states" not "test equality for identical states". It is not a blocker for me, though.

Reply:
I almost believed you here :)
Identity is between the objects, not the dict it gets as input.
There is no special handling for identical dicts, the data from the dict is extracted to attributes and the attributes values are checked for equality.